src/com/android/calendar/month/MonthWeekEventsView.java

```diff
@@ -1525,11 +1548,24 @@ protected void drawEventRectangle(Canvas canvas, int day)  {
             mBoundaries.setRectangle(mFormat.getDaySpan(day), mFormat.getEventLines());
             mEventSquarePaint.setStyle(getRectanglePaintStyle());
             mEventSquarePaint.setColor(getRectangleColor());
-            canvas.drawRect(r, mEventSquarePaint);
+
+            if(!mEvent.allDay){
+                r.left+=EVENT_RECT_MARGIN_LEFT;
+                r.right+=EVENT_RECT_MARGIN_LEFT;
```

Comment:
logic here sounds unusual, you add marin_left whereas in line 1559 you substract margin right

Reply:
Yes, it is a bit unusual. It seems that the background version behaves differently for those cases, they seem to be differently sized, that is why they have different margins (and why the right margin is added instead of substracted) should i add a comment there why that is in the code?